objects/src/accounts/storage/mod.rs

```diff
@@ -283,6 +279,18 @@ impl AccountStorage {
             storage_map.apply_delta(map_delta)?;
         }
 
+        // --- update storage slots -------------------------------------------
+        // this will also update roots of updated storage maps, and thus should be run after we
+        // update storage maps - otherwise the roots won't match
+
+        for &slot_idx in delta.cleared_items.iter() {
```

Comment:
This already happened, I think. Because we trigger a `set_item` in MASM with the new `StorageRoot`. So there is no need to do it again here.

Reply:
I've also added [this code](https://github.com/0xPolygonMiden/miden-base/blob/ec24a37a38618d12d735f805c77f2c5fddc15bd6/objects/src/accounts/storage/mod.rs#L329) which ensures that slots corresponding to storage maps can be set only to the roots of these maps. So, if we try to update the roots first (before updating the maps themselves), the assertion would get triggered.